src/streamlink_cli/argparser.py

```diff
@@ -781,13 +781,7 @@ def build_parser():
         Default is 3.
         """
     )
-    transport.add_argument(
-        "--hls-segment-stream-data",
-        action="store_true",
-        help="""
-        Immediately write segment data into output buffer while downloading.
-        """
-    )
+    transport.add_argument("--hls-segment-stream-data", help=argparse.SUPPRESS)
```

Comment:
you still need `action="store_true"`
otherwiser it will break userscripts/configs

---

cli

```
streamlink: error: argument --hls-segment-stream-data: expected one argument
```
config file

```
[cli][debug]  --hls-segment-stream-data=--ffmpeg-ffmpeg=/usr/bin/ffmpeg
```